Content.Server/GameObjects/Components/Disposal/DisposalRouterComponent.cs

```diff
@@ -0,0 +1,175 @@
+﻿using Content.Server.Interfaces;
+using Content.Server.Interfaces.GameObjects.Components.Items;
+using Content.Shared.GameObjects.EntitySystems;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects.Components.UserInterface;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Maths;
+using Robust.Shared.ViewVariables;
+using System;
+using System.Collections.Generic;
+using static Content.Shared.GameObjects.Components.Disposal.SharedDisposalRouterComponent;
+
+namespace Content.Server.GameObjects.Components.Disposal
+{
+    [RegisterComponent]
+    [ComponentReference(typeof(IActivate))]
+    [ComponentReference(typeof(IDisposalTubeComponent))]
+    public class DisposalRouterComponent : DisposalJunctionComponent, IActivate
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IServerNotifyManager _notifyManager;
+        [Dependency] private readonly ILocalizationManager _localizationManager;
+#pragma warning restore 649
+        public override string Name => "DisposalRouter";
+
+        [ViewVariables]
+        private BoundUserInterface _userInterface;
+
+        [ViewVariables]
+        private HashSet<string> _tags;
+
+        [ViewVariables]
+        public bool Anchored =>
+            !Owner.TryGetComponent(out CollidableComponent collidable) ||
+            collidable.Anchored;
+
+        public override Direction NextDirection(DisposalHolderComponent holder)
+        {
+            var next = Owner.Transform.LocalRotation.GetDir();
+            var directions = ConnectableDirections();
+
+            if (holder.Tags.Overlaps(_tags))
+            {
+                return directions[1];
+            }
+
+            return next;
+        }
+
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            _userInterface = Owner.GetComponent<ServerUserInterfaceComponent>()
+                .GetBoundUserInterface(DisposalRouterUiKey.Key);
+            _userInterface.OnReceiveMessage += OnUiReceiveMessage;
```

Comment:
`.CloseAll()` needs to be called on `_userInterface` in this component's OnRemove method.